test/test_compile.py

```diff
@@ -0,0 +1,97 @@
+import argparse
+import os.path
+import pytest
+
+from otk.command import compile
+
+
+@pytest.mark.parametrize(
+    "parser, arguments, dry_run, input_data, output_data, sys_exit_code, log_message",
```

Comment:
I wonder if this one can be split a bit more, it feels a bit hard to read and the test body has a bunch of "if" conditions inside, could this be split into multiple smaller tests maybe somehow? Maybe a "happy" and "sad" might already help?